### PR TITLE
Implement client metrics

### PR DESCRIPTION
**Pull request by davidliu**
Protocol PR: https://github.com/livekit/protocol/pull/821

### CODE REVIEW

File: .changeset/curly-trains-fix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"client-sdk-android": patch
```
Comment (bcherry) [inline .changeset/curly-trains-fix.md:2]: minor?

File: livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt [hunk 2]
```diff
@@ -1028,6 +1028,20 @@ internal constructor(
...
+
+            LivekitModels.DataPacket.ValueCase.CHAT_MESSAGE -> {
+                // TODO
+            }
+
+            LivekitModels.DataPacket.ValueCase.RPC_REQUEST,
+            LivekitModels.DataPacket.ValueCase.RPC_ACK,
+            LivekitModels.DataPacket.ValueCase.RPC_RESPONSE,
+            -> {
+                // TODO
```
Comment (bcherry) [inline livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt:1043]: there's nothing to actually do, right?

File: livekit-android-sdk/src/main/java/io/livekit/android/room/metrics/RTCMetricsManager.kt [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+import kotlin.coroutines.resume
+
+suspend fun collectMetricsTracking(room: Room, rtcEngine: RTCEngine) = coroutineScope {
+    launch { collectPublisherMetrics(room, rtcEngine) }
+    launch { collectSubscriberMetrics(room, rtcEngine) }
+}
+
+suspend fun collectPublisherMetrics(room: Room, rtcEngine: RTCEngine) {
+    while (currentCoroutineContext().isActive) {
+        delay(1000)
```
Comment (bcherry) [inline livekit-android-sdk/src/main/java/io/livekit/android/room/metrics/RTCMetricsManager.kt:46]: add an explanatory comment?

**Review by bcherry — APPROVED** [review]

File: livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt (hunk 2 above)
Comment (davidliu) [reply livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt:1043]: Nothing to do in this PR, but there is client work required in the future for these data messages types right?

**Pull request merged**